frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -1,223 +1,243 @@
-  FormGroup,
-  FormHelperText,
-  HelperText,
-  HelperTextItem,
-  Label,
-  LabelGroup,
-  Modal,
-  TextInput,
-} from '@patternfly/react-core';
-import { ExclamationCircleIcon } from '@patternfly/react-icons';
-import DashboardDescriptionListGroup, {
-  DashboardDescriptionListGroupProps,
-} from './DashboardDescriptionListGroup';
-
-type EditableTextDescriptionListGroupProps = Partial<
-  Pick<DashboardDescriptionListGroupProps, 'title' | 'contentWhenEmpty'>
-> & {
+import React, { useState } from 'react';
+import { Label, LabelGroup, Alert, AlertVariant, Button } from '@patternfly/react-core';
+import DashboardDescriptionListGroup from './DashboardDescriptionListGroup';
+
+interface EditableLabelsProps {
   labels: string[];
-  saveEditedLabels: (labels: string[]) => Promise<unknown>;
-  allExistingKeys?: string[];
+  onLabelsChange: (labels: string[]) => Promise<void>;
   isArchive?: boolean;
-};
+  title?: string;
+  contentWhenEmpty?: string;
+  allExistingKeys: string[];
+}
 
-const EditableLabelsDescriptionListGroup: React.FC<EditableTextDescriptionListGroupProps> = ({
+export const EditableLabelsDescriptionListGroup: React.FC<EditableLabelsProps> = ({
   title = 'Labels',
   contentWhenEmpty = 'No labels',
   labels,
-  saveEditedLabels,
+  onLabelsChange,
   isArchive,
-  allExistingKeys = labels,
+  allExistingKeys,
 }) => {
-  const [isEditing, setIsEditing] = React.useState(false);
-  const [unsavedLabels, setUnsavedLabels] = React.useState(labels);
-  const [isSavingEdits, setIsSavingEdits] = React.useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isSavingEdits, setIsSavingEdits] = useState(false);
+  const [unsavedLabels, setUnsavedLabels] = useState(labels);
 
-  const editUnsavedLabel = (newText: string, index: number) => {
-    if (isSavingEdits) {
+  const validateLabels = (labelsList: string[]): string[] => {
+    const errors: string[] = [];
+
+    const duplicatesMap = new Map<string, number>();
+    labelsList.forEach((label) => {
+      duplicatesMap.set(label, (duplicatesMap.get(label) || 0) + 1);
+    });
+
+    const duplicateLabels: string[] = [];
+    duplicatesMap.forEach((count, label) => {
+      if (count > 1) {
+        duplicateLabels.push(label);
+      }
+    });
+
+    if (duplicateLabels.length > 0) {
+      if (duplicateLabels.length === 1) {
+        errors.push(
+          `**${duplicateLabels[0]}** already exists. Ensure that each label is unique and does not match any existing property key.`,
+        );
+      } else {
+        const lastLabel = duplicateLabels.pop();
+        const formattedLabels = duplicateLabels.map((label) => `**${label}**`).join(', ');
+        errors.push(
+          `${formattedLabels} and **${lastLabel}** already exist. Ensure that each label is unique and does not match any existing property key.`,
+        );
+      }
+    }
+
+    labelsList.forEach((label) => {
+      if (!labelsList.includes(label) && allExistingKeys.includes(label)) {
```

Comment:
This logic is wrong. `!labelsList.includes(label)` will always be false here because we're looping over labelsList, the label will always be in labelsList. So this validation case is never hit, and I am allowed to create a label that already exists as a key in the properties table below, which replaces it and loses that property value.

I think instead, we need to either consider `allExistingKeys` when we are setting up the `duplicatesMap` or we need to have this logic be somehow `if (weDontAlreadyHaveAnErrorForThisLabel && allExistingKeys.includes(label))`. In that case, the message could be specific to the fact that the label matches an existing property key, like `**${label}** already exists as a property key. Ensure that each label is unique and does not match any existing property key.`